src/main/java/util/Parser.java

```diff
@@ -0,0 +1,119 @@
+package util;
+
+import alanExceptions.*;
+
+// Enum for input types
+enum InputType {
+    mark,
+    unmark,
+    todo,
+    deadline,
+    event,
+    delete,
+    find,
+}
+
+/**
+ * This class is used to parse command line inputs.
+ */
+public class Parser {
+    /**
+     * Parses command line input to ParsedData.
+     *
+     * @param type Type of input.
+     * @param input The input.
+     * @return ParsedData.
+     * @throws AlanException The exception in case of failure.
+     */
+    public ParsedData parse(InputType type, String input) throws AlanException {
+        ParsedData result;
+        switch (type) {
+
+            case todo:
+                result = parseUntimedTask(input);
+                break;
+            case deadline:
+            case event:
+                result = parseTimedTask(input);
+                break;
+            case find:
+                result = parseFind(input);
+                break;
+            default:
+                //for now default is for mark and unmark
+                result = parseListMod(input);
+        }
+
+        return result;
+    }
+
+    private ParsedData parseFind(String input) throws AlanException {
+        String first[] = input.split(" ", 2);
+        String command = first[0];
+        String keyword;
+
+        try {
+            keyword = first[1];
+        } catch (IndexOutOfBoundsException e) {
+            throw new NoKeywordException(command);
+        }
+        return new ParsedData(keyword);
+    }
+
+    private ParsedData parseListMod(String input) throws AlanException {
+        int listIndex;
+        String[] first = input.split(" ", 2);
+        String command = first[0];
+
+        try {
+            listIndex = Integer.parseInt(first[1]);
+        } catch (IndexOutOfBoundsException | NumberFormatException e) {
+            throw new NoValueException(command);
+        }
+
+
+        return new ParsedData(listIndex - 1);
+    }
+
+    private ParsedData parseTimedTask(String input) throws AlanException {
+        String command, task, during, time;
+        String[] first, second, third;
```

Comment:
Trivial issue but can consider using clearer variable names here to increase readability of code 😃